pennylane/transforms/zx/converter.py

```diff
@@ -0,0 +1,302 @@
+
+    outputs = []
+    for mapper in (q_mapper, c_mapper):
+        for label in mapper.labels():
+            qubit = mapper.to_qubit(label)
+            vertex = graph.add_vertex(VertexType.BOUNDARY, qubit, row)
+            outputs.append(vertex)
+            pre_vertex = mapper.prev_vertex(label)
+            graph.add_edge(graph.edge(pre_vertex, vertex))
+
+    graph.set_inputs(tuple(inputs))
+    graph.set_outputs(tuple(outputs))
+
+    return graph
+
+
+def from_zx(graph, split_phases=True):
+    """It converts a graph from PyZX to a PennyLane tape, if graph is diagram-like.
+    Args:
+        graph(Graph): ZX graph in PyZX.
+        split_phases(bool): If True the phases are split.
+    """
+
+    # Avoid to make PyZX a requirement for PennyLane.
+    try:
+        # pylint: disable=import-outside-toplevel, unused-import
+        import pyzx
+        from pyzx.utils import EdgeType, VertexType, FloatInt
+        from pyzx.graph.base import VT
+
+    except ImportError as Error:
+        raise ImportError(
+            "This feature requires PyZX. It can be installed with: pip install pyzx"
+        ) from Error
+
+    # List of PennyLane operations
+    operations = []
+
+    qubits = graph.qubits()
+    graph_rows = graph.rows()
+    types = graph.types()
+
+    # Parameters are phases in the ZX framework
+    params = graph.phases()
+    rows = {}
+
+    inputs = graph.inputs()
+
+    # Set up the rows dictionary
+    for vertex in graph.vertices():
+        if vertex in inputs:
+            continue
+        row_index = graph.row(vertex)
+        if row_index in rows:
+            rows[row_index].append(vertex)
+        else:
+            rows[row_index] = [vertex]
+
+    for row_key in sorted(rows.keys()):
+        for vertex in rows[row_key]:
+
+            qubit_1 = qubits[vertex]
+            param = params[vertex]
+            type_1 = types[vertex]
+            neighbors = [w for w in graph.neighbors(vertex) if graph_rows[w] < row_key]
+
+            # The graph is not diagram like.
+            if len(neighbors) != 1:
+                raise qml.QuantumFunctionError(
+                    "Graph doesn't seem circuit like: multiple parents. Use extract circuit function."
+                )
+
+            neighbor_0 = neighbors[0]
+
+            if qubits[neighbor_0] != qubit_1:
+                raise qml.QuantumFunctionError(
+                    "Graph doesn't seem circuit like: cross qubit connections"
+                )
+
+            if graph.edge_type(graph.edge(neighbor_0, vertex)) == EdgeType.HADAMARD:
+                operations.append(qml.Hadamard(wires=qubit_1))
+
+            if type_1 == VertexType.BOUNDARY:
+                continue
+
+            # Given the phase add a 1 qubit gate
+            if param != 0 and not split_phases:
+                if type_1 == VertexType.Z:
+                    param = np.pi * float(param)
+                    operations.append(qml.RZ(param, wires=qubit_1))
```

Comment:
```suggestion
                    scaled_param = np.pi * float(param)
                    operations.append(qml.RZ(scaled_param, wires=qubit_1))
```
Given we're dispatching based on `param` and it is controlling logic flow, we should re-use the variable name.